runelite-client/src/main/java/net/runelite/client/plugins/itemskeptondeath/ItemsKeptOnDeathPlugin.java

```diff
@@ -299,65 +326,128 @@ else if (keepCount > 0)
 			// 2) is under the deep wilderness line
 			// 3) is outside of the wilderness, or item has a broken form
 			if (!Pets.isPet(id)
-				&& !isTradeable(c) && wildyLevel <= DEEP_WILDY
+				&& !LostIfNotProtected.isLostIfNotProtected(id)
+				&& !isTradeable(itemManager.getItemComposition(id)) && wildyLevel <= DEEP_WILDY
 				&& (wildyLevel <= 0 || BrokenOnDeathItem.isBrokenOnDeath(i.getId())))
 			{
-				final Widget itemWidget = createItemWidget(kept, itemQuantity, c);
-				itemWidget.setOnOpListener(ScriptID.DEATH_KEEP_ITEM_EXAMINE, 1, itemQuantity, c.getName());
-				keptItems.add(itemWidget);
+				keptItems.add(new TempItem(id, qty));
 			}
 			else
 			{
 				// Otherwise, the item is lost
-				final Widget itemWidget = createItemWidget(lost, itemQuantity, c);
-				itemWidget.setOnOpListener(ScriptID.DEATH_KEEP_ITEM_EXAMINE, 0, itemQuantity, c.getName());
-				lostItems.add(itemWidget);
+				lostItems.add(new TempItem(id, qty));
 			}
 		}
 
-		int rows = (keptItems.size() + MAX_ROW_ITEMS - 1) / MAX_ROW_ITEMS;
-		// Show an empty row if there isn't anything
-		if (rows > 0)
+		if (hasClueBox)
 		{
-			// ORIGINAL_LOST_Y/HEIGHT includes a row already
-			rows--;
+			boolean alreadyProtectingClue = false;
+			for (final TempItem item : keptItems)
+			{
+				if (isClueBoxable(item.getId()))
+				{
+					alreadyProtectingClue = true;
+					break;
+				}
+			}
+
+			if (!alreadyProtectingClue)
+			{
+				int boxedId = -1;
+				// Clue box protects the last clue in your inventory so loop over the players inv
+				for (final Item i : inv)
+				{
+					final int id = i.getId();
+					if (id != -1 && isClueBoxable(id))
+					{
+						boxedId = id;
+					}
+				}
+
+				if (boxedId != -1)
+				{
+					// Move the boxed item to the kept items container and remove it from the lost items container
+					for (final TempItem boxableItem : lostItems)
+					{
+						if (boxableItem.getId() == boxedId)
+						{
+							if (boxableItem.getQty() > 0)
+							{
+								boxableItem.setQty(boxableItem.getQty() - 1);
+								keptItems.add(new TempItem(boxedId, 1));
+							}
+							else
+							{
+								lostItems.remove(boxableItem);
```

Comment:
There is no test coverage of this? This will throw a concurrent modification exception